DataSetExplorer/ConsoleUI.cs

```diff
@@ -0,0 +1,293 @@
+            {
+                result = _dataSetCreationService.CreateDataSetSpreadsheet(projectName.ToString(), projects[projectName].ToString());
+                Console.WriteLine(result.ToString());
+            }
+        }
+
+        private static ListDictionary GetProjectsForDataSet()
+        {
+            ListDictionary projects = new ListDictionary();
+            string projectName;
+            string projectAndCommitUrl;
+            string finishOption;
+
+            Console.WriteLine("\nProjects for data set:");
+            do
+            {
+                projectName = GetAnswerOnQuestion("Enter project name: ");
+                projectAndCommitUrl = GetAnswerOnQuestion("Enter project and commit URL: ");
+                projects.Add(projectName, projectAndCommitUrl);
+                finishOption = GetAnswerOnQuestion("Finished? (y/n): ");
+            } while (finishOption.Equals("n"));
+
+            return projects;
+        }
+
+        private static void AnalyzeDataSet()
+        {
+            string dataSetPath;
+            string outputPath;
+            string finishOption;
+
+            do
+            {
+                dataSetPath = GetAnswerOnQuestion("Enter data set folder path: ");
+                outputPath = GetAnswerOnQuestion("Enter output folder path: ");
+                ChooseAnalyzeDataSetOption(dataSetPath, outputPath);
+                finishOption = GetAnswerOnQuestion("Finished? (y/n): ");
+            } while (finishOption.Equals("n"));
+        }
+
+        private static void ChooseAnalyzeDataSetOption(string dataSetPath, string outputPath)
+        {
+            string chosenOption;
+            do
+            {
+                WriteAnalyzeDataSetOptions();
+                chosenOption = GetAnswerOnQuestion("Your option: ");
+
+                Result<string> result;
+                switch (chosenOption)
+                {
+                    case "1":
+                        result = _dataSetAnalysisService.FindInstancesRequiringAdditionalAnnotation(dataSetPath, outputPath);
+                        Console.WriteLine(result.ToString());
+                        break;
+                    case "2":
+                        result = _dataSetAnalysisService.FindInstancesWithAllDisagreeingAnnotations(dataSetPath, outputPath);
+                        Console.WriteLine(result.ToString());
+                        break;
+                    case "x":
+                        break;
+                    default:
+                        ChosenOptionErrorMessage(chosenOption);
+                        break;
+                }
+            } while (!chosenOption.Equals("x"));
+        }
+
+        private static void ExportDataSet()
+        {
+            var projects = GetAnnotatedProjects();
+            var annotators = GetAnnotators();
+            IDataSetExporterService dataSetExportationService = new DataSetExportationService(new FullDataSetFactory(projects, annotators));
```

Comment:
Services should be initialized during Console UI construction. They should also be stateless, holding references only to those classes that do the work (because they are coordinators).

The FullDataSetFactory is probably also stateless, and if it is not, it should be instantiated for each request (by the service) so that the Service remains stateless. The projects and annotators should be supplied as input if they are inputs.